server/routes/profile.js

```diff
@@ -1,25 +1,36 @@
 const express = require('express');
 
-const createProfileRouter = () => {
+const createProfileRouter = ({ offenderService }) => {
   const router = express.Router();
 
+  const getPersonalisation = async user => {
+    const events = await offenderService.getEventsForToday(user);
+
+    const timetableError = events?.error;
+    const { morning, afternoon, evening } = events || {};
+    const signedInUser = user.getFullName();
+
+    return {
+      signedInUser,
+      events: { error: timetableError, morning, afternoon, evening },
+    };
+  };
+
   router.get('/', async (req, res, next) => {
     try {
-      const templateParams = {
+      const { user } = req;
+      const personalisation = user ? await getPersonalisation(user) : {};
```

Comment:
Not one for now but worth contemplating, we could potentially have some snazzy personalisation middleware to decouple the routing code and the personalisation code.